pyaedt/edb_core/EDB_Data.py

```diff
@@ -1595,14 +1605,32 @@ def rotation(self):
         str
             Value for the rotation.
         """
-        pad_values = self._padstack_methods.GetPadParametersValue(self._edb_padstack, self.layer_name, self.pad_type)
-        return pad_values.Item5.ToString()
+
+        pad_values = self._edb_padstack.GetData().GetPadParametersValue(
+            self.layer_name, self.int_to_pad_type(self.pad_type)
+        )
+        return pad_values[5].ToString()
 
     @rotation.setter
     def rotation(self, rotation_value):
 
         self._update_pad_parameters_parameters(rotation=rotation_value)
 
+    @pyaedt_function_handler()
+    def int_to_pad_type(self, val=0):
+        """Convert an integer to an EDB.PadGeometryType.
```

Comment:
Can this be written out in English, such as: Convert an integer to an EDB pad geometry type.

Reply:
PadGeometryType is a class in Edb